myapp/app/helpers/sessions_helper.rb

```diff
@@ -22,4 +26,11 @@ def require_login
     flash[:danger] = I18n.t('need_login')
     redirect_to login_path
   end
+
+  def require_admin_login
+    return if admin_logged_in?
+
+    flash[:danger] = I18n.t('need_admin_login')
```

Comment:
This message tells "The admin page is here!" to bad guys, so is a good idea to make it a 404 error.

Reply:
@zackey-y That's right....!!

I fixed that! https://github.com/Fablic/training/commit/a0d74b49a172d7e90f3e16ad209b02328134d433